src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,251 @@
+     * months in title case. In some cases, the time or date may be omitted, upon which
+     * the parsing rules are as follows:
+     *
+     * 1. If only time is supplied, parse as upcoming datetime with the specified time.
+     * 2. If no time is supplied, parse as upcoming datetime at midnight.
+     * 3. If year is not supplied, parse as upcoming datetime with the specified time/day/month.
+     * 4. If date and year supplied, parsed as specified datetime.
+     *
+     * e.g. if datetime string is "13 Jan" and the current date is "25 Jan 2021",
+     * this will be parsed as "13 Jan 2022, 00:00".
+     *
+     * Available date formats:
+     * - 13 Sep
+     * - 13 Sep 2020
+     * - 13 September
+     * - 13 September 2020
+     * - 13/9
+     * - 13/9/2020
+     * - Sep 13
+     * - September 13
+     * - 2020-09-13
+     *
+     * Available time formats:
+     * - 9:23
+     * - 23:09
+     *
+     * Available datetime formats:
+     * - 13 Sep 9:03
+     * - 13/9 9:03
+     * - 9:03 13 Sep
+     * - 9:03 13/9/2020
+     *
+     * @param input Datetime string.
+     * @return LocalDateTime.
+     * @throws DukeExceptionIllegalArgument When parsing fails.
+     */
+    public static LocalDateTime parseDate(String input) throws DukeExceptionIllegalArgument {
+        return parseDate(input, LocalDateTime.now());
+    }
+
+    /**
+     * Parses datetime string relative to supplied datetime.
+     *
+     * Method created primarily for mocking. See parsing rules for parseDate.
+     *
+     * @param input Datetime string.
+     * @param now LocalDateTime.
+     * @return LocalDateTime.
+     * @throws DukeExceptionIllegalArgument When parsing fails.
+     */
+    public static LocalDateTime parseDate(String input, LocalDateTime now) throws DukeExceptionIllegalArgument {
+        input = input.strip();
+        now = now.withSecond(0).withNano(0);
+        String inputAddYear = input+"_"+now.getYear();
+
+        for (int i = 0; i < DATE_FORMATS.length; i++) {
+            LocalDateTime date;
+            DateTimeFormatter fmt = DATE_FORMATS[i];
+
+            // No date supplied
+            if (i == 0) {
+                LocalTime readout;
+                try {
+                    readout = LocalTime.parse(input, fmt);
+                } catch (DateTimeParseException e) {
+                    continue;
+                }
+                date = now.withHour(readout.getHour()).withMinute(readout.getMinute());
+                if (date.isBefore(now)) {
+                    date = date.plusDays(1);
+                }
+            }
+
+            // No time supplied, assume midnight
+            else if (i <= 5) {
+                LocalDate readout;
+                try {
+                    readout = LocalDate.parse(input, fmt);
+                } catch (DateTimeParseException e) {
+                    continue;
+                }
+                date = readout.atTime(0, 0);
+            }
```

Comment:
Should the else if be on the same line as the close brace? Occurs a few times below too

Reply:
Yep, fixed